registry/remote/repository_test.go

```diff
@@ -7447,3 +7447,24 @@ func TestRepository_do(t *testing.T) {
 		t.Errorf("Repository.do() = %v, want %v", gotWarnings, wantWarnings)
 	}
 }
+
+func TestRepository_clone(t *testing.T) {
+	repo, err := NewRepository("localhost:1234/repo/image")
+	if err != nil {
+		t.Fatalf("invalid repository: %v", err)
+	}
+
+	crepo := repo.clone()
+
+	if repo.Reference != crepo.Reference {
+		t.Fatal("references should be the same")
+	}
+
+	if !reflect.DeepEqual(&repo.referrersPingLock, &crepo.referrersPingLock) {
+		t.Fatal("referrersPingLock should be copied")
+	}
+
+	if !reflect.DeepEqual(&repo.referrersMergePool, &crepo.referrersMergePool) {
+		t.Fatal("referrersMergePool should be copied")
+	}
```

Comment:
Should verify other fields as well?

Reply:
I do not think that is worth doing.  `clone()` is too simple in that respect.  If you wanted to fix this problem for real you would need to change your definition of `RepositoryOptions` to not be a `Repository` [hack](https://github.com/ktarplee/oras-go/blob/1325b43bb11915e7bdf88b0228dcfdce90b14000/registry/remote/registry.go#L35), then you can embed `RepositoryOptions` in a `Repository` and then there would be no issue with forgetting to clone fields because you would clone the entire `RepositoryOptions` and ignore the rest (possibly) which would contain the Mutex among other fields.